### PR TITLE
Final milestone

### PR DESCRIPTION
**Pull request by sumancrest0001**
Feature of the milestone.
1. Added functions on ButtonPanel and App.js file to pass props.
2. Updated readme file

### CODE REVIEW

File: src/logic/operate.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      result = firstOperand.minus(secondOperand).toString();
+      break;
+
+    case ('+'):
+      result = firstOperand.add(secondOperand).toString();
+      break;
+
+    case ('%'):
+      result = firstOperand.times(secondOperand.div('100')).toString();
+      break;
```
Comment (bolah2009) [inline src/logic/operate.js]: The percentage symbol should be a modifier like the `-/+` symbol, where `85% === 8.5`.

| Current behaviour | Recommended behaviour |
| :-----: | :-----: |
| percentage-bug | percentage-fix |



> `operate` switches `operation` and uses `plus`, `minus`, `times`, `div` and `modifier` (turning the number into a percentage based number, `e.g. 95 % == .95`) functions ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone4/performOperate.png))

File: src/logic/calculate.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        total: '0', operation: null, next: null, isCalculated: true,
+      };
+
+
+    default:
+      if (!operation && !isCalculated) {
+        return {
+          total: total === null ? name : total + name,
+          operation,
+          next,
```
Comment (bolah2009) [inline src/logic/calculate.js]: Let's handle cases when the button clicked is `.` differently. 
Multiple `.` should not be allowed. That is, when the value contains a dot, adding another dot should be ignored and not allowed.

See below:

dot-bug

File: src/logic/operate.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
+/*eslint-disable*/
```
Comment (bolah2009) [inline src/logic/operate.js]: Is there a reason why the whole eslint is disabled for this file? :thinking:

**Review by bolah2009 — CHANGES_REQUESTED** [review]
Awesome job @sumancrest0001 👏🏾👏🏾👏🏾

Please check out the comments below and make changes accordingly, also note that you don't need to change comments marked with `optional` tag.

#### :recycle:  Make these changes and submit another code review request. 👍🏾

#### Feel free to reach me on slack if you having issues implementing these features. :relaxed: 

Good luck with your next submission. :star2: 

Cheers
[Bola](https://bolabuari.com/)

###### PS: Note that comments tagged as `OPTIONAL` are not part of the minimum requirements to approve this PR

File: src/components/ButtonPanel.js [hunk 4]
```diff
@@ -1,95 +1,121 @@
...
           color="#ebe8e8"
           name="AC"
         />
         <Button
+          clickHandler={handleClick}
           color="#ebe8e8"
           name="+/-"
         />
         <Button
+          clickHandler={handleClick}
```
[the remark this answers is not in the document]
Comment (sumancrest0001) [reply src/components/ButtonPanel.js]: Thank you for the feedback. I have implemented it.

File: src/logic/operate.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+      result = firstOperand.times(secondOperand.div('100')).toString();
+      break;
+
+    case ('X'):
+      result = firstOperand.times(secondOperand).toString();
+      break;
+
+    case ('÷'):
+      result = firstOperand.div(secondOperand).toString();
+      break;
```
[the remark this answers is not in the document]
Comment (sumancrest0001) [reply src/logic/operate.js]: done

File: src/logic/operate.js (hunk 1 above)
Comment (sumancrest0001) [reply src/logic/operate.js]: implemented it.

File: src/logic/operate.js (hunk 3 above)
Comment (sumancrest0001) [reply src/logic/operate.js]: It was unintended. Sorry for this.

**Comment by sumancrest0001** [comment]
Features added.
1. Covered all the suggested edge cases.
2. Removed unintended eslint disable comment.

File: src/components/ButtonPanel.js [hunk 6]
```diff
@@ -1,95 +1,143 @@
...
           name="="
         />
       </div>
     </div>
-  );
-}
+  ); */
```
Comment (MauricioRobayo) [inline src/components/ButtonPanel.js]: Please remove the commented code.

**Review by MauricioRobayo — CHANGES_REQUESTED** [review]
Well done fixing previous issues, but... 😣 seems like you left some commented code after you made the last changes.

* [ ] It’s better to delete commented code. If you need to review a previous version of your code, use git history ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

Status: __Request changes__.

File: src/components/ButtonPanel.js (hunk 6 above)
Comment (sumancrest0001) [reply src/components/ButtonPanel.js]: Sorry it was a silly mistake

File: src/components/Button.js [hunk 7]
```diff
@@ -3,21 +3,31 @@ import PropTypes from 'prop-types';
 import classes from './Button.module.css';
 
 const button = props => {
-  const { name, wide, color } = props;
+  const {
+    name, wide, color, clickHandler,
+  } = props;
```
[the remark this answers is not in the document]
Comment (sumancrest0001) [reply src/components/Button.js]: thank you

**Comment by sumancrest0001** [comment]
Removed all the commented code

**Review by shloch — APPROVED** [review]
## Great job bro....thumbs up 😇😇😎😎

### we're good to go...you can merge💯

File: src/components/App.js [hunk 8]
```diff
@@ -3,16 +3,26 @@ import Helmet from 'react-helmet';
...
+
 class App extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      total: '30',
+      total: null,
+      operation: null,
+      next: null,
+      isCalculated: false,
```
Comment (sumancrest0001) [inline src/components/App.js:17]: this is a state

**Pull request merged**